src/python/dxpy/scripts/dx.py

```diff
@@ -3202,7 +3214,8 @@ def run(args):
 
     handler = try_call(get_exec_handler, args.executable, args.alias)
 
-    if args.depends_on and isinstance(handler, dxpy.DXWorkflow):
+    if args.depends_on and \
+            (isinstance(handler, dxpy.DXWorkflow) or isinstance(handler, dxpy.DXGlobalWorkflow)):
```

Comment:
This is more crucial for `dx run` commit which is coming next so I'll just leave it here.